File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector < int > & nums) {
        int max_sum =   INT_MIN     ,
            sum     =   0           ,
            n       =   nums.size() ;

        for(int i = 0; i < n; i++){
            sum += nums[i];
            if(sum > max_sum)
                max_sum = sum;
            if(sum < 0)
                sum = 0;
        }
        return max_sum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = -1000000001
        currentSum = 0
        
        for i in range(len(nums)):
            currentSum += nums[i]
            if currentSum > maxSum:
                maxSum = currentSum
            if currentSum < 0: 
                currentSum = 0
                
        return maxSum